application/backend/server/src/paste/manager.rs

```diff
@@ -0,0 +1,133 @@
+use crate::app::state::AppState;
+use crate::common::ResponseInfo;
+use crate::common::{session::validate_token, InternalServerError, PasteData};
+use actix_web::http::header;
+use actix_web::{delete, get, post, web, HttpRequest, HttpResponse, Responder};
+use log::info;
+use serde::{Deserialize, Serialize};
+
+#[derive(Deserialize)]
+struct AddPasteReq {
+    text: String,
+}
+
+#[derive(Serialize)]
+struct GetPasteResp {
+    text: String,
+}
+
+#[post("/paste")]
+pub async fn add_paste(
+    state: web::Data<AppState>,
+    add_paste_req: web::Json<AddPasteReq>,
+    req: HttpRequest,
+) -> impl Responder {
+    info!("Inserting paste.");
+
+    let result = save_paste(&state, &add_paste_req, &req).await;
+    match result {
+        Ok(_) => {
+            info!("Paste added successfully.");
+            HttpResponse::Created().json({
+                ResponseInfo {
+                    info: "Paste added successfully.".to_string(),
+                }
+            })
+        }
+        Err(e) => e.handle_error_for_http_resp(),
+    }
+}
+
+async fn save_paste(
+    state: &web::Data<AppState>,
+    add_paste_req: &web::Json<AddPasteReq>,
+    req: &HttpRequest,
+) -> Result<(), InternalServerError> {
+    check_session(&req)?;
+
+    let client_resp = state.get_key_client().get_key().await?;
+    let paste_data = PasteData {
+        key: client_resp,
+        content: add_paste_req.text.clone(),
+    };
+
+    state.get_db().add_paste(&paste_data).await
+}
+
+fn check_session(req: &HttpRequest) -> Result<(), InternalServerError> {
```

Comment:
function validates token extracted from header so its name could imply that it authenticates user
also should be defined in some http auth module to be reusable for different endpoints

Reply:
Majority of common code is separated to common part in: application/backend/server/src/common/session.rs. If further reasue will be needed in future this part will be refactored.

Reply:
why not fix now and allow new features to be introduced without refactoring?